staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -263,3 +360,46 @@ type CustomResourceSubresourceScale struct {
 	// +optional
 	LabelSelectorPath *string `json:"labelSelectorPath,omitempty" protobuf:"bytes,3,opt,name=labelSelectorPath"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConversionReview describes a conversion request/response.
+type ConversionReview struct {
+	metav1.TypeMeta `json:",inline"`
+	// Request describes the attributes for the conversion request.
+	// +optional
+	Request *ConversionRequest `json:"request,omitempty" protobuf:"bytes,1,opt,name=request"`
+	// Response describes the attributes for the conversion response.
+	// +optional
+	Response *ConversionResponse `json:"response,omitempty" protobuf:"bytes,2,opt,name=response"`
+}
+
+// ConversionRequest describes a conversion request parameters.
+type ConversionRequest struct {
+	// UID is an identifier for the individual request/response. It allows us to distinguish instances of requests which are
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
+	// The UID is meant to track the round trip (request/response) between the KAS and the WebHook, not the user request.
+	// It is suitable for correlating log entries between the webhook and apiserver, for either auditing or debugging.
+	// +optional
+	UID types.UID `json:"uid,omitempty" protobuf:"bytes,1,opt,name=uid"`
+	// The version to convert given object to. E.g. "stable.example.com/v1"
+	APIVersion string `json:"apiVersion" protobuf:"bytes,2,name=apiVersion"`
+	// Object is the CRD object to be converted.
+	Object runtime.RawExtension `json:"object" protobuf:"bytes,3,name=object"`
+	// IsList indicates that this is a List operation and Object contains a list of items.
+	IsList bool `json:"isList" protobuf:"varint,4,name=isList"`
+}
+
+// ConversionResponse describes a conversion response.
+type ConversionResponse struct {
+	// UID is an identifier for the individual request/response.
+	// This should be copied over from the corresponding AdmissionRequest.
+	// +optional
+	UID types.UID `json:"uid,omitempty" protobuf:"bytes,1,opt,name=uid"`
+	// ConvertedObject is the converted version of request.Object if the Result is nil.
+	// +optional
+	ConvertedObject runtime.RawExtension `json:"convertedObject" protobuf:"bytes,2,name=convertedObject"`
+	// Result contains extra details into why a conversion request was failed. If it is not nil, it means
```

Comment:
it's not a pointer. It can't be nil.

Reply:
please make it a pointer & optional

Reply:
runtime.RawExtension can be nil (having nil value). The optional runtime.RawExtension is not a pointer in other APIs (at least the ones I checked like WatchEvent or AdmissionRequest)